### PR TITLE
feat(Badge): add badge component

### PR DESCRIPTION
**Pull request by cathysiller**
JIRA: SURF-441

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

# Badges
badges-helixui

### CODE REVIEW

**Comment by araiford** [comment]
:shipit: LGTM

File: docs/components/badges/index.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title: Badges
+also:
```
Comment (CITguy) [inline docs/components/badges/index.html]: Remove empty `also`

File: docs/components/badges/index.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+---
+{% extends 'component.njk' %}
+{% block content %}
+  <section>
+    <h2 id="link-badges">Link Badges</h2>
+    <div class="demo">
+      <a href="">Link</a><span class="hxBadge">1</span>
```
Comment (CITguy) [inline docs/components/badges/index.html]: Shouldn't the badge be inside the `<a>`?

File: docs/components/badges/index.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    </div>
+    {% code 'html' %}
+      <a href="">Link</a><span class="hxBadge">1</span>
+    {% endcode %}
+  </section>
+
+  <section>
+    <h2 id="icon-badges">Icon Badges</h2>
+    <div class="demo">
+      <hx-icon type="bell"></hx-icon><span class="hxBadge"></span>
```
Comment (CITguy) [inline docs/components/badges/index.html]: Did you ever figure out the issue with the extra space when the badge is on a separate line?

File: docs/components/badges/index.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      <hx-icon type="bell"></hx-icon><span class="hxBadge">99+</span>
+    {% endcode %}
+  </section>
+
+  <section>
+    <h2 id="tab-badges">Tab Badges</h2>
+    <div class="demo">
+      <hx-tabset>
+        <hx-tablist>
+          <hx-tab>First<span class="hxBadge">1</span></hx-tab>
```
Comment (CITguy) [inline docs/components/badges/index.html]: Can you update the markup to add some whitespace and nesting?

```html
<hx-tab>
  First
  <span class="hxBadge">1</span>
</hx-tab>
```

File: docs/components/badges/index.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          <hx-tabpanel>...</hx-tabpanel>
+          <hx-tabpanel>...</hx-tabpanel>
+          <hx-tabpanel>...</hx-tabpanel>
+        </hx-tabcontent>
+      </hx-tabset>
+    </div>
+    {% code 'html' %}
+      <hx-tabset>
+        <hx-tablist>
+          <hx-tab>First<span class="hxBadge">1</span></hx-tab>
```
Comment (CITguy) [inline docs/components/badges/index.html]: See above comment about whitespace and nesting.

File: docs/components/badges/index.html [hunk 6]
```diff
@@ -0,0 +1,107 @@
+---
+title: Badges
+---
+{% extends 'component.njk' %}
+{% block content %}
+  <section>
+    <h2 id="link-badges">Link Badges</h2>
+    <div class="demo">
+      <a href="">Link<span class="hxBadge">1</span></a>
```
Comment (CITguy) [inline docs/components/badges/index.html]: Please nest elements for legibility.

```html
<a href="">
  Link
  <span class="hxBadge">1</span>
</a>
```

File: docs/components/badges/index.html [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    </div>
+    {% code 'html' %}
+      <a href="">Link<span class="hxBadge">1</span></a>
```
Comment (CITguy) [inline docs/components/badges/index.html]: see above comment

File: docs/_data/nav.json5 [hunk 8]
```diff
@@ -68,6 +69,7 @@
             { label: '<hx-accordion>', path: 'hx-accordion' },
             { label: '<hx-accordion-panel>', path: 'hx-accordion-panel' },
             { label: '<hx-alert>', path: 'hx-alert' },
+            { label: '<hx-badge>', path: 'hx-badge' },
```
Comment (nimbinatus) [inline docs/_data/nav.json5]: Zoom: Remove as this is just a custom class, not an element.

File: src/helix-ui/styles/components/badges.less [hunk 9]
```diff
@@ -0,0 +1,21 @@
+.hxBadge {
+  align-items: center;
+  background-color: @red-900;
+  border-radius: 1em;
+  color: @gray-0;
+  display: inline-flex;
+  height: 1.25em;
+  justify-content: center;
+  margin-left: 0.25rem;
```
Comment (nimbinatus) [inline src/helix-ui/styles/components/badges.less]: Remove this line.

File: src/helix-ui/styles/components/badges.less [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+  min-width: 1.25em;
+  padding: 0 0.25em;
+  text-align: center;
+}
+
+hx-icon + .hxBadge {
+  font-size: 0.625rem;
+  margin-left: -0.75rem;
+  min-width: 0.75rem;
```
Comment (nimbinatus) [inline src/helix-ui/styles/components/badges.less]: Check to remove?

**Pull request merged**